torch/__init__.py

```diff
@@ -62,22 +62,27 @@
     else:
         cuda_path = ''
 
-    if sys.version_info >= (3, 8):
-        dll_paths = list(filter(os.path.exists, [th_dll_path, py_dll_path, nvtoolsext_dll_path, cuda_path]))
+    kernel32 = ctypes.WinDLL('kernel32.dll', use_last_error=True)
+    dll_paths = list(filter(os.path.exists, [th_dll_path, py_dll_path, nvtoolsext_dll_path, cuda_path]))
 
-        for dll_path in dll_paths:
+    for dll_path in dll_paths:
+        if sys.version_info >= (3, 8):
             os.add_dll_directory(dll_path)
-
-    if is_conda or sys.version_info < (3, 8):
-        dll_paths = [th_dll_path, py_dll_path, nvtoolsext_dll_path, cuda_path]
-        dll_paths = list(filter(os.path.exists, dll_paths)) + [os.environ['PATH']]
-
-        os.environ['PATH'] = ';'.join(dll_paths)
+        else:
+            res = kernel32.AddDllDirectory(dll_path)
+            if res == 0:
+                err = ctypes.WinError(ctypes.get_last_error())
+                err.strerror += ' Error adding "{}" to the DLL directories.'.format(dll_path)
+                raise err
 
     import glob
     dlls = glob.glob(os.path.join(th_dll_path, '*.dll'))
     for dll in dlls:
-        ctypes.CDLL(dll)
+        res = kernel32.LoadLibraryExW(dll, 0, 0x00001100)
```

Comment:
Same as above here:
```
    if hasattr('kernel32', LoadLibraryExW'):
        res = kernel32.LoadLibraryExW(dll, 0, 0x1100)
    else:
        ctypes.CDLL(dll)
````

Reply:
well, the function itself is supported even in xp but the arguments isn't. I'll try to look for ways to check that.